### DISCUSSION TITLE
Linting with ESLint and Prettier, I get errors from start with create-next-app

### DISCUSSION THREAD

**Discussion by dimitri-bourreau** [vercel/next.js#16475] [Help] [answered]
Hi everyone, these are my first steps in Next.js, thanks in advance for your help. 🙏

**Issue**
I ran ``npx create-next-app`` and I tried to install some linting plugins.
Unfortunately I ran into errors. I could fix them following advices online.

Apart fixing, I updated no files. I still get 2 errors with ``eslint .``:
```bash
/project-path/next-project/pages/_app.js
  3:18  error  'Component' is missing in props validation  react/prop-types
  3:29  error  'pageProps' is missing in props validation  react/prop-types
```

I followed some advices online, and I can't find others struggling like I do so... I must have been wrong somewhere! Can you help me?

**Details**
[repo](https://github.com/tobudim/next-blog) / [commit with everything I done after create-next-app](https://github.com/tobudim/next-blog/commit/30780bd44818d213cbdac9dab5f522a96b8039ca)

Output: terminal
Editor: VIM

*package.json*
```
         "dependencies": {
		"next": "9.5.2",
		"react": "16.13.1",
		"react-dom": "16.13.1",
		"save-dev": "0.0.1-security" <-- Oops, gotta fix this btw
	},
	"devDependencies": {
		"eslint": "^7.7.0",
		"eslint-config-prettier": "^6.11.0",
		"eslint-plugin-jsx-a11y": "^6.3.1",
		"eslint-plugin-prettier": "^3.1.4",
		"eslint-plugin-react": "^7.20.6",
		"eslint-plugin-react-hooks": "^4.1.0",
		"eslint-plugin-simple-import-sort": "^5.0.3",
		"prettier": "^2.0.5"
	}
```

*eslintrc.json*
```json
{
	"root": true,
	"settings": {
		"react": {
			"version": "detect"
		}
	},
	"env": {
		"amd": true,
		"browser": true,
		"commonjs": true,
		"es6": true,
		"node": true
	},
	"extends": [
		"eslint:recommended",
		"plugin:react/recommended",
		"plugin:jsx-a11y/recommended",
		"plugin:prettier/recommended"
	],
	"globals": {
		"Atomics": "readonly",
		"SharedArrayBuffer": "readonly"
	},
	"parserOptions": {
		"sourceType": "module",
		"ecmaFeatures": {
			"jsx": true
		},
		"ecmaVersion": 11
	},
	"plugins": ["simple-import-sort", "prettier", "react"],
	"rules": {
		"prettier/prettier": ["error", {}, { "usePrettierrc": true }],
		"react/react-in-jsx-scope": "off",
		"simple-import-sort/sort": "error",
		"jsx-a11y/anchor-is-valid": [
			"error",
			{
				"components": ["Link"],
				"specialLink": ["hrefLeft", "hrefRight"],
				"aspects": ["invalidHref", "preferButton"]
			}
		]
	},
	"globals": { "React": "writable" }
}
```

**Comment by jamesmosier** [comment, marked as answer]
Hi @tobudim. You can fix this 1 of 2 ways:

1. In the `rules` section of your `eslintrc.json` you could remove that lint rule: `"react/prop-types": "off"`
2. The "error" is coming from the lack of prop-type definitions in `_app.js`. If you wanted to add prop-type validation, you could install the [prop-types](https://www.npmjs.com/package/prop-types) package via `npm install prop-types --save-dev` and then use it in `_app.js` like this:

```js
import PropTypes from "prop-types";
import "../styles/globals.css";

function MyApp({ Component, pageProps }) {
  return <Component {...pageProps} />;
}

MyApp.propTypes = {
  pageProps: PropTypes.shape({}),
  Component: PropTypes.elementType 
};

export default MyApp;

```

**Reply by dimitri-bourreau** [reply]
Thanks a lot @jamesmosier !
I can see how simple it is, I do appreciate you took the time and the humility to help me out kindly. Much appreciated.

**Reply by Vanderson7593** [reply]
Thanks! @jamesmosier
